### DISCUSSION TITLE
MIT18.01/18.02: Calculus - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#124] [Announcements]
# MIT18.01/18.02: Calculus - CS自学指南

CS自学指南

https://csdiy.wiki/%E6%95%B0%E5%AD%A6%E5%9F%BA%E7%A1%80/MITmaths/

**Comment by scatyf3** [comment]
补充一下，油管数学网红在可汗学院有讲过多元微积分，有视觉化视频讲解，note和帮助理解概念的习题：https://www.khanacademy.org/math/multivariable-calculus；对于数学小白可能比这个MIT的方便入门

**Reply by meitiandouyaozuoreaideshi** [reply]
这课程没啦

**Reply by Tina-LT** [reply]
油管上还有哦

**Reply by kenyx6** [reply]
还有吧https://www.khanacademy.org/math/multivariable-calculus

**Reply by scatyf3** [reply]
kenyx6补充的这个连接是可以打开的，我的连接不知道为啥挂了（btw有这么多人回复，小透明真的很开心

**Reply by zerohk** [reply]
因为中文被识别成链接的一部分了

**Reply by JasonZhaoWW** [reply]
看可汗学院的是不是不用再看18：02了

**Comment by sudongyuer** [comment]
想请问一下大佬们，可以不学数学么，本身对数学不感兴趣，而且底子特别差

**Reply by wuzhenqing** [reply]
我觉得可能不太行，微积分，线性代数，概率论的基础还是很重要的，否则后续的课程用到的数学知识可能看不懂了

**Reply by wuzhenqing** [reply]
数学底子差是可以慢慢弥补的，没有谁的基础知识是与生俱来的

**Reply by therealemery** [reply]
底子差才更应该好好补

**Reply by LukaDonciic** [reply]
数学如果只是应用的话时间可以弥补一切，不用学那么深奥，理解会用就行

**Reply by kolihhan** [reply]
想问数学底子差有什么推荐的学习顺序呢？

**Comment by ColoCoffee** [comment]
大佬们有没有英语的课程推荐/(ㄒoㄒ)/~~;还有大概啥水平能听懂这些课啊?6级够么?

**Reply by wuzhenqing** [reply]
英语菜鸡瑟瑟发抖，六级还没有过。但是有字幕的话还是能听懂这些课的。

**Reply by CY-from-away** [reply]
我觉得在国内的话大家的英语除非是本专业，一般很少有上课提升的，我个人来讲高考之后就只是在看各种美剧和电影还有油管视频，没有刻意锻炼，然后后来才发现原来这几年英语有变好很多。语言学习这方面兴趣是最好的老师，我觉得你可以去找些娱乐性质的东西看看，被动输入一段时间，想不会都难了。

**Reply by ColoCoffee** [reply]
> 我觉得在国内的话大家的英语除非是本专业，一般很少有上课提升的，我个人来讲高考之后就只是在看各种美剧和电影还有油管视频，没有刻意锻炼，然后后来才发现原来这几年英语有变好很多。语言学习这方面兴趣是最好的老师，我觉得你可以去找些娱乐性质的东西看看，被动输入一段时间，想不会都难了。

谢谢你的建议😁，这样和课程结合确实效果很好

**Reply by WwWangGuan** [reply]
emmm网易公开课上有中文字幕版本，我觉得还不错

**Reply by JzzzX** [reply]
YouTube上有中字

**Reply by opetizer** [reply]
主要在于你觉得过了六级对这个有没有帮助。。我没过四级的时候就开始看了，熟练了英语的环境就没那么难了，相信自己。

**Reply by AlexLiu-C** [reply]
感觉应该问题不大, 但是理解数学本身就很难, 使用外语理解数学的话会更加难
要不要试试看中文字幕?

**Reply by CxHsin** [reply]
我记得cs61abc是个好用的路线，学英语方面也是

**Comment by opetizer** [comment]
课程视频加载不出来是什么原因

**Reply by TwilightRainy** [reply]
你翻墙了吗？

**Reply by lihanyinuo** [reply]
https://xfltd.win/#/register?code=ETtmSjlz 用过最好的翻墙软件，又便宜又稳定，10元120g永久有效 够我用个一年半载了

**Comment by eli-yip** [comment]
请问，18.01A和18.01SC有什么区别呢？

**Reply by eli-yip** [reply]
已经解决，谢谢作者。
考虑到自学，我还是上SC吧。

**Reply by WwWangGuan** [reply]
能跟我说说吗？

**Reply by eli-yip** [reply]
SC课程设置更全面更细节，更贴近全自学者。
具体的可以google一下what's the difference between  mit 18.01 and 18.01sc，quora上有个回答讲的很好，或者mitopencourseware的简介上讲的也很清楚。

**Reply by WwWangGuan** [reply]
谢谢

**Comment by Lwanna-be-better** [comment]
请问为什么课程视频无法打开呢？

**Reply by YasonGF** [reply]
因为是外网，所以一般访问不了

**Reply by TwilightRainy** [reply]
作者有关于翻墙的教程，可以看一下

**Reply by abigail22222** [reply]
需要翻墙

**Reply by bili2046** [reply]
机场可以来

**Comment by elpsykongloo** [comment]
请问各位是不是不用太学微积分？感觉考个3.7就算了

**Comment by Danxfj** [comment]
感谢北大的大佬，希望我学了这里的东西也能去北大

**Reply by asukagaze** [reply]
加油

**Comment by teacup418** [comment]
在MITocw搜18.01有两个结果，分别是托管在ocw的[18.01sc](https://ocw.mit.edu/courses/18-01sc-single-variable-calculus-fall-2010/)和托管在MIT open learning library的[18.01](https://openlearninglibrary.mit.edu/courses/course-v1:MITx+18.01.1x+2T2019/about)，应该选哪一个？有什么异同？

**Reply by MOSSJD** [reply]
我用的是前一个, 感觉很不错

**Comment by teacup418** [comment]
B站有字幕组申请了3b1b的中国代理，现在不用代理也能看3b1b的视频啦！[点此观看微积分的本质](https://www.bilibili.com/video/BV1qW411N7FU/)

**Reply by teacup418** [reply]
准确的说是不用梯子也能看视频啦

**Comment by WwWangGuan** [comment]
好耶！&nbsp;


(?˙?˙?)
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年10月10日(星期一) 下午3:33
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PKUFlyingPig/cs-self-learning] MIT18.01/18.02: Calculus - CS自学指南 (Discussion #124)





 
B站有字幕组申请了3b1b的中国代理，现在不用代理也能看3b1b的视频啦！点此观看微积分的本质

**Comment by chenccQAQ** [comment]
有中文课程以代替吗。。。？只过了四级直接啃英文教程有一点难

**Reply by teacup418** [reply]
高考英语过了100分其实就可以看懂了……不过你想要中文可以去b站看看有没有搬运，或者上网易公开课应该有翻译。

**Comment by tuyula** [comment]
有学习群么, 有就拉我, 没有加我微tuyu550574002, 我来创建

**Comment by ColoCoffee** [comment]
谢谢你的建议

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: YANG ZYU ***@***.***>
发送时间: Friday, November 18, 2022 12:17:03 AM
收件人: PKUFlyingPig/cs-self-learning ***@***.***>
抄送: ColoCoffee ***@***.***>; Comment ***@***.***>
主题: Re: [PKUFlyingPig/cs-self-learning] MIT18.01/18.02: Calculus - CS自学指南 (Discussion #124)


我本科是英文专业的，但读了一个计算机的研究生。然而，尽管有较多专业语言训练（四六级裸考也有650/630），但我之前看数学和计算机的英文视频时也有不少困难，特别是很多“简单”的术语对从小用中文接受基础教育的我来说非常陌生，而英专大量的课程都是文学、语言学相关的“高级”词汇，对此帮助很有限。大一、大二的时候，老师要求我们定期汇报自己看的“英语原声”内容，同学们多数都报告了什么BBC、CNN等新闻媒体；可我当时真心对这些不感兴趣，于是我选择了 Khan Academy 的数学系列视频作为我的课外英语学习素材――我当时甚至是从幼儿园级别的内容开始看的。尽管久别英专后，我的英语水平下降很多，但这段经历对我的帮助非常大，也许你也可以试一试――建议学一遍 Khan Academy 的 High School 数学部分（可以开英语字幕），然后再根据本文的建议去学习相关的大学级别的数学课程，我相信此时的你的英语能力（以及英文环境下的数学基础）一定是没有问题的！

―

**Comment by TwilightRainy** [comment]
没有学习过高中数学，可以直接学习微积分课程吗？

**Reply by dghtucs** [reply]
可以，相关性不大

**Comment by wuzhenqing** [comment]
虽然说微积分的课程内容，比如极限，数列这些知识，有初中的基础就可以学习了，但是建议还是学习完高中的一些知识，比如三角函数，解析几何之后再学微积分。更宝贵的东西是逻辑思维，是一种可以学习各种公理化的知识的能力。


武震卿
***@***.***

哈尔滨工业大学


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PKUFlyingPig/cs-self-learning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月25日(星期五) 下午3:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PKUFlyingPig/cs-self-learning] MIT18.01/18.02: Calculus - CS自学指南 (Discussion #124)





 
没有学习过高中数学，可以直接学习微积分课程吗？

**Comment by Baymax-Rob** [comment]
大多数课程都可以在b站上找到，直接搜索MIT18.01等等

**Comment by ghost** [comment]
这个18.01最好到edX上面去看，配套随堂练习和作业，b站搬运只有视频。英语不难，带字幕放0.75倍速完全跟得上。
此外edX还有一个摸底测试，如果成绩没达到它的要求，就需要回头补补高中数学知识，或者找门precalculus课来上。

**Reply by ghost** [reply]
再补充一点：edX还有助教，回复非常及时，讲解也十分细致。虽说问题需要用英文描述，但借助翻译软件完全应付得来。最后别忘了感谢，“在外”学习要体现出我们的礼貌。

**Reply by kiwitreezhu** [reply]
我在edx没有搜到这个课，不知是什么原因，可以麻烦提供一下链接吗，感恩！

**Reply by Loukin-SSL** [reply]
要注册账号后使用
https://learning.edx.org/course/course-v1:MITx+18.01.1x+2T2017/home

**Reply by kiwitreezhu** [reply]
uu我想问下你怎么搜到的，我在edx半天搜不到这个课。。。

**Reply by Loukin-SSL** [reply]
谷歌直接搜“edx 18.01”

**Reply by Fengxn0609** [reply]
找不到 edx18.06, 大佬可以提供下链接吗？

**Reply by happytech-web** [reply]
问下大家edx版本和csdiy上给的链接版本有啥区别，听哪个好呢

**Reply by GHCRIOUS** [reply]
这个edx unit0没问题的 后面unit1开始题目就没有了 
mit自己的是正常的 https://openlearninglibrary.mit.edu/courses/course-v1:MITx+18.01.1x+2T2019/course/

**Comment by soubuniaola** [comment]
作者你好，为啥18.02的note找不到啊

**Reply by MOSSJD** [reply]
有的, 只不过18.02sc大部分课程视频不再有note版了, 只有worked example还有一些作业

**Reply by steinsgate97122** [reply]
痛心，18.02居然没有note，只能慢慢看视频了

**Comment by sdyg0315** [comment]
补充一下 这个可以使用前苏联卓里奇的数学分析 这个教材比较经典
课程链接https://www.youtube.com/channel/UCeAIkvDFeHwAIiQBFHP/videohttp

**Comment by jason-hue** [comment]
我能不能只学MIT的这个课程，期末能过不，没有时间先把宋浩的听一遍再来听这个

**Reply by wzr222** [reply]
不行，这个比较简单可能过不了。如果关注期末的话，最好看你们的往年试卷

**Comment by Joye-Yu** [comment]
18.02的课真的很棒，在讲到梯度场gradient field的时候“**内心直接卧槽！！！原来如此**”，有些physicist‘s concept还是从物理学角度来学吧，而不是用纯数学，那一部分给我开天眼了一样哈哈哈哈

**Comment by XIAOYAO828** [comment]
学了三年的法律，已经三年没有碰过数学了，求教各位大佬，有没有课程对文科生友好的，从最最基础的教的

**Reply by FeiNiaoBF** [reply]
看看3Blue1Brown 的微积分的本质,看完可以试着看看mit的

**Reply by treya-lin** [reply]
3年并不久。从mit微积分开始即可。

**Comment by AlexanderWang888** [comment]
有想一起把mit18.0x 数学这一个体系搞下来的没，加我qq：2581606394；人多了我拉群

**Comment by dghtucs** [comment]
讨论入群780716950

**Comment by jason-hue** [comment]
有离散数学的课程嘛？我想窥见本质

**Comment by my-little-republic** [comment]
18.01sc的problem set 我真的好困惑，比如 https://ocw.mit.edu/courses/18-01sc-single-variable-calculus-fall-2010/resources/mit18_01sc_pset2prb/ 这个里面的第一题，说by using (2) 可是我也没找到（2）是啥啊？每次想做练习题都感觉和worked example差别好大 题目经常没头没尾的（内心暴走）

**Reply by MOSSJD** [reply]
这里面需要用到问题集，你看看旁边有没有下载按钮

**Reply by ffxpmu** [reply]
解决了吗

**Reply by zbnzbj** [reply]
我也是😭

**Reply by treya-lin** [reply]
> 解决了吗

一直没搞明白。视频跟完了，但是没有做问题集。

**Comment by mwh040709** [comment]
学了这个还用再学同济版高等数学吗

**Reply by AENSTERM** [reply]
同问 你现在有想法了吗

**Reply by MOSSJD** [reply]
我学完了，个人感觉不用

**Comment by AENSTERM** [comment]
大家好，我是一名想转专业的大一新生。我发现mit微积分的课程和学校的微积分课程在进度上有些不一样，我是应该跟着学校的进度走吗？

**Reply by lliqa** [reply]
跟学校进度保证过考试，用mit课程加深理解

**Reply by MOSSJD** [reply]
可以跟也可以不跟，我没有跟，最后学的内容差不多

**Comment by CS-CN-MrFresh** [comment]
编程为什么要学微积分啊🤔

**Reply by Otgavin** [reply]
编程离不开算法呀，想要更好的算法就要用到数学知识

**Reply by GuMengSama** [reply]
你学的是计算机科学，而不是编程，只会编程你又比培训班出来的强多少呢

**Reply by Caesar-Victory** [reply]
看到计算机自学指南包含微积分，而有些人觉得不需要学习数据，我才知道为什么叫做计算机科学与技术

**Comment by KafkaJR** [comment]
想问下各位大佬，18.01sc有讲义吗？

**Comment by ccckkk567** [comment]
想问下各位大佬，在大一学过一遍后，学完大概要多久，想提前规划下时间

**Comment by JoshLLin** [comment]
note怎么找啊

**Reply by hideonbush0111** [reply]
18.01每节课上都有 Accompanying Notes (PDF)，这个就是

**Comment by cloud107** [comment]
一定要看这个课吗，国内的课能不能应付考试❓

**Reply by wuzhenqing** [reply]
看这些课程是为了学更多的东西。如果只是为了应付考试，B站上就有挺多突击课程的。

**Comment by 17797310028** [comment]
有没有大佬推荐一些数学分析的课捏

**Reply by 1010fzp0101** [reply]
陶哲轩的实分析很不错！

**Comment by Floating02** [comment]
想问问考研学这两门课可以吗

**Comment by khalil-pu** [comment]
请问一下所谓的notes就是那些pdf吗

**Reply by Otgavin** [reply]
网站的每个pdf不都有resource type说明吗

**Reply by khalil-pu** [reply]
找到了，就是那些Lecture Notes,我还以为是文字形式的，没想到是pdf

**Comment by t1nD1** [comment]
为什么我翻墙了，视频也打不开，就是黑屏状态

**Comment by yzyloveschickenthigh** [comment]
推荐一本多元微积分的教材
Vector Calculus - Susan Jane Colley
比18.02更细致，例题和练习也更多，可以作为补充

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287

**Comment by ShengqinJiang** [comment]
问一下微积分在算法方面的应用主要是哪个方面啊

**Reply by Caesar-Victory** [reply]
ChatGPT5一下

**Reply by Caesar-Victory** [reply]
微积分在算法中的应用广泛且多样，主要体现在以下几个方面：

1. **优化算法**：
   微积分常用于优化问题中，尤其是在寻找最小值或最大值时。例如，梯度下降法就是利用导数来寻找函数的局部最小值。计算损失函数的导数并沿着负梯度方向更新参数，是机器学习和深度学习中的核心技术。

2. **连续优化问题**：
   在算法设计中，很多问题可以被建模为连续优化问题。通过微积分中的梯度、二阶导数（Hessian矩阵）等工具，可以高效地求解最优解。比如最小二乘法就是通过微积分来最小化误差的平方和。

3. **动态系统模拟与控制**：
   微积分也用于模拟和控制动态系统的行为，特别是在求解常微分方程（ODEs）时。许多算法，例如卡尔曼滤波器、粒子滤波器等，都涉及到微积分的知识，用于跟踪和预测系统状态。

4. **数值积分和近似**：
   在离散化问题中，微积分提供了数值积分的方法，如梯形法、辛普森法等，广泛用于工程计算、科学计算和机器学习中的数据逼近。

5. **概率分布与统计**：
   微积分在概率论和统计学中也有着重要应用，例如在计算概率密度函数的期望、方差和协方差时，通常需要使用积分来求解这些期望值。

总的来说，微积分帮助算法处理连续变化的问题，优化系统性能，推导出精确的数学模型，并提高算法的效率与稳定性。

**Comment by OLeslieO** [comment]


**Reply by yousamar** [reply]
蹲

**Reply by MayBe112358** [reply]
cy

**Comment by badgirl306** [comment]
这个对数学有要求吗，我是说我的数学只有初中水平可以学吗
